### PR TITLE
add ansible role, template

### PR DESCRIPTION
**Pull request by sallyom**
@damemi @abhgupta
This PR depends on https://github.com/openshift/online/pull/1637

To test, have to create a builder image in a devenv:
$ ssh ec2-user@blah
$ sudo -i
$ cd /data/src/github.com/openshift; git clone https://github.com/sallyom/online-console-extensions.git; git checkout ansible-template; cd online-console-extensions/s2i-custom-builder
$ docker build -t online-console-extensions-builder:latest .

Edit: I've added the docker build custom-builder tasks to the role here, until the builder image is synced to a repo somewhere.  
Run the devenv-launch.yml  playbook from the above PR (#1637) to see the online-console-extensions in action.

### CODE REVIEW

File: ansible/roles/oso_console_extensions/files/console-extensions-template.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        uri: ${GIT_REPO}
+        ref: ${GIT_REF}
+    strategy:
+      sourceStrategy:
+        env:
+        - name: TARGET
+          value: ${TARGET}
+        from:
+          kind: DockerImage
+          name: ${NAME}-builder:latest
```
Comment (sallyom) [inline ansible/roles/oso_console_extensions/files/console-extensions-template.yml]: For now, to test this, I've created this builder image in the devenv.

File: ansible/roles/oso_console_extensions/templates/console-extensions-template.yml.j2 [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      - imageChangeParams:
+          automatic: true
+          containerNames:
+          - ${NAME}
+          from:
+            kind: ImageStreamTag
+            name: ${NAME}:latest
+            namespace: test
+        type: ImageChange
+        imageChangeParams:
```
Comment (Miciah) [inline ansible/roles/oso_console_extensions/templates/console-extensions-template.yml.j2]: You have two `imageChangeParams` stanzas for the same trigger (only difference being that the first one has `.from.namespace`).

File: ansible/roles/oso_console_extensions/templates/console-extensions-template.yml.j2 [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+          automatic: true
+          containerNames:
+            - ${NAME}
+          from:
+            kind: ImageStreamTag
+            name: ${NAME}:latest
+      - type: ConfigChange
```
Comment (Miciah) [inline ansible/roles/oso_console_extensions/templates/console-extensions-template.yml.j2]: You list two `ConfigChange` triggers.

File: ansible/roles/oso_console_extensions/templates/console-extensions-template.yml.j2 (hunk 2 above)
Comment (sallyom) [reply ansible/roles/oso_console_extensions/templates/console-extensions-template.yml.j2]: @Miciah thanks for reviewing this!! Much needed

File: ansible/roles/oso_console_extensions/tasks/main.yml [hunk 4]
```diff
@@ -5,9 +5,42 @@
...
+# Install (docker-py) python package.
+- pip:
+    name: docker-py
+
 - import_tasks: uninstall.yml
   when: oso_ext_uninstall | bool
 
+- debug: msg="Building {{ oso_ext_builder_image}} builder image from {{ oso_ext_builder_repo }} ref {{ oso_ext_builder_git_ref }}"
+
+- name: 'Check for extistence of repo {{ oso_ext_builder_repo }}'
```
Comment (Miciah) [inline ansible/roles/oso_console_extensions/tasks/main.yml]: "existence"

File: ansible/roles/oso_console_extensions/tasks/main.yml [hunk 5]
```diff
@@ -5,9 +5,42 @@
...
+  git:
+    repo: "{{ oso_ext_builder_repo }}"
+    version: "{{ oso_ext_builder_git_ref }}"
+    accept_hostkey: true
+    dest: /data/src/github.com/openshift/httpd-container
+    update: no
+  # Git may not be installed on remote hosts.
+  delegate_to: "{{ inventory_hostname }}"
```
Comment (Miciah) [inline ansible/roles/oso_console_extensions/tasks/main.yml]: Did you mean `delegate_to: localhost`?

If Git is not installed on the remote host, how can you use it in the next step to fetch the Git repository?

File: ansible/roles/oso_console_extensions/files/console-extensions-template.yml (hunk 1 above)
Comment (sallyom) [reply ansible/roles/oso_console_extensions/files/console-extensions-template.yml]: EDIT: I've added the building of the custom-builder into the devenv playbook, so to test, run the online devenv playbook.

File: ansible/roles/oso_console_extensions/tasks/main.yml (hunk 4 above)
Comment (sallyom) [reply ansible/roles/oso_console_extensions/tasks/main.yml]: :) thanks

File: ansible/roles/oso_console_extensions/tasks/main.yml (hunk 5 above)
Comment (sallyom) [reply ansible/roles/oso_console_extensions/tasks/main.yml]: yea,  i removed that comment.  We can add 'Git' to prerequisites, i guess.  Also, delegate_to should be "{{ inventory_hostname }}" (this role should be run in OSEv3 host), bc the git clone happens in the OSEv3 host, not local.

Comment (Miciah) [reply ansible/roles/oso_console_extensions/tasks/main.yml]: But `inventory_hostname` is the name of the remote host, which is where the task runs by default (if neither `delegate_to` nor `local_action` is specified), right? So if I understand correctly, the `delegates_to: "{{ inventory_hostname }}"` stanza is a no-op and can be omitted.

**Review by damemi — APPROVED** [review]
/lgtm

**Pull request merged**